src/main/java/PeselValidator.java

```diff
@@ -0,0 +1,126 @@
+public class PeselValidator {
+
+
+    public boolean validate(String pesel){
+        return isLengthCorrect(pesel)
+                        && isFormatCorrect(pesel)
+                        && isMonthCorrect(pesel)
+                        && isDayCorrect(pesel)
+                        && isControlDigitCorrect(pesel);
+    }
+
+
+    public boolean validate(String pesel, String gender){
+        return validate(pesel) && whatGender(pesel).equals(gender);
+    }
+
+    private boolean isLengthCorrect(String pesel) {
+
+        if (pesel.length() == 11) return true;
+        else return false;
+    }
+
+
+    private boolean isMonthCorrect(String pesel) {
+
+        int monthNumber = Integer.valueOf(pesel.substring(2,4));
+
+        if (monthNumber > 0 && monthNumber < 13) return true;
+        if (monthNumber > 20 && monthNumber < 33) return true;
+        if (monthNumber > 40 && monthNumber < 53) return true;
+        if (monthNumber > 60 && monthNumber < 73) return true;
+        if (monthNumber > 80 && monthNumber < 93) return true;
+        else return false;
+    }
+
+
+    private String whatGender(String pesel) {
+
+        char genderDigit = pesel.charAt(9);
+
+        if (genderDigit == 0
```

Comment:
to check if digit is odd/even you could use modulo operator:
if(genterDigit%2 == 0) (...)